File: stone_lib/src/main/java/com/github/klee0kai/stone/annotations/component/GcSoftScope.java
package com.github.klee0kai.stone.annotations.component;


import javax.inject.Scope;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * A standard library scope that lists all cached objects using soft references.
 * Used for garbage collection and caching change methods.
 */

@GcScopeAnnotation
@Scope
@Retention(value = RetentionPolicy.CLASS)
@Target(value = ElementType.METHOD)
public @interface GcSoftScope {
}

File: stone_multiplatform/src/commonMain/kotlin/stone/annotations/component/GcSoftScope.kt
package stone.annotations.component


import javax.inject.Scope

/**
 * A standard library scope that lists all cached objects using soft references.
 * Used for garbage collection and caching change methods.
 */
@stone.annotations.component.GcScopeAnnotation
@javax.inject.Scope
@Retention(AnnotationRetention.BINARY)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class GcSoftScope 
